.github/workflows/Test.yml

```diff
@@ -313,13 +313,74 @@ jobs:
     - name: '🚧 Test package'
       run: GHDL=ghdl ./testsuite/testsuite.sh ${{ matrix.suite }}
 
+#
+# pyGHDL DOM Test
+#
+
+  DOM-test:
+    needs: win-build
+    runs-on: windows-latest
+    strategy:
+      fail-fast: false
+      max-parallel: 8
+      matrix:
+        include: [
+          {icon: '🟦', installs: 'MINGW32', arch: i686,   pkg: 'mcode' },
+          #{icon: '🟦', installs: "MINGW32", arch: i686,   pkg: "llvm"  }, ! Not yet functional
+          #{icon: '🟪', installs: "MINGW64", arch: x86_64, pkg: "mcode" }, ! mcode is not yet supported on win64
+          {icon: '🟪', installs: 'MINGW64', arch: x86_64, pkg: 'llvm'  },
+        ]
+    name: '${{ matrix.icon }} DOM · ${{ matrix.installs }} · ${{ matrix.pkg }} · ${{ matrix.suite }}'
+    defaults:
+      run:
+        shell: msys2 {0}
+    steps:
+
+    - name: '${{ matrix.icon }} Setup MSYS2'
+      uses: msys2/setup-msys2@v2
+      with:
+        msystem: ${{ matrix.installs }}
+        update: true
+        install: >
+          mingw-w64-${{ matrix.arch }}-python-pip
+          mingw-w64-${{ matrix.arch }}-python-setuptools
+
+    - name: '⚙️ git config'
+      run: git config --global core.autocrlf input
+      shell: bash
+
+    - name: '🧰 Checkout'
+      uses: actions/checkout@v2
+
+    - name: '📥 Download artifact: package'
+      uses: actions/download-artifact@v2
+
+    - name: '🚧 Install package and Python dependencies'
+      run: |
+        pacman --noconfirm -U artifact/mingw-w64-${{ matrix.arch }}-ghdl-${{ matrix.pkg }}-*.zst
+        pip3 install .
+
+    - name: Test the DOM and pretty print all VHDL sources
+      run: |
+        cat > test_pyGHDL.sh <<EOF
+        #!/usr/bin/env bash
```

Comment:
This should `set -e` or equivalent to avoid swallowing the exit code.

Reply:
Yeah, but that is on purpose. I don't think it makes sense to enforce it if the DOM.py scripts handles exception as follows: https://github.com/ghdl/ghdl/blob/master/pyGHDL/cli/DOM.py#L53-L54. So, currently, it wouldn't be meaningful. We know it fails a lot because the feature is not complete. 

We expect multiple changes in the following days, since @Paebbels is working really hard on this feature. This job is for having a quick coarse feedback about the progress. Then, we will enhance this for properly reporting how many files were successful, which failed, etc.

In fact, `set -e` should be the default if the call is not wrapped (which is just for aesthetic purposes). This is the MSYS2 entrypoint: https://github.com/msys2/setup-msys2/blob/master/main.js#L169. However, see https://github.com/umarcor/ghdl/runs/2857691698?check_suite_focus=true. It didn't fail on the first error.

Reply:
Since several crashes were fixed yesterday and today, I updated the PR for using pytest, instead of a bash wrapper.